packages/fuels-accounts/src/provider/retryable_client.rs

```diff
@@ -63,6 +63,11 @@ impl RetryableClient {
             .await
     }
 
+    pub async fn submit_and_await_commit(&self, tx: &Transaction) -> io::Result<TransactionStatus> {
```

Comment:
Maybe we should turn this into our `TxStatus`? If it is to be used by us further up the call tree?